### ISSUE TITLE
Building without egl

### ISSUE THREAD

**Issue by matheusmonte** [WebPlatformForEmbedded/WPEWebKit#454]
This is more a question than a issue.

My target device doesn't have a EGL support, so I'm trying to compile without this dependence.

Is someone already do this? Can you explain or share some experience on this?

Thanks for your time

**Comment by blino** [comment]
Do you have OpenGL ?

I am working on getting WPE to work without any EGL/OpenGL.
So far, I am able to build WPE without EGL/OpenGL on a MIPS platform without GPU, and get pages loaded and running, but with no graphical output on screen.
I plan to add support for a cairo backend over DirectFB, inside WebKit.

But the patches are not so pretty, to handle the non-accelerated DrawingArea.

**Comment by matheusmonte** [comment]
Hi @blino Thanks so much to your time! 

It's exactly what I will start to try to do. Are you working on some branch/repository that I can clone and help you?

Thanks again, I will wait your answer.

**Comment by matheusmonte** [comment]
@blino Do you have more information about this? A link? A blog post?

**Comment by blino** [comment]
Hi,

I have cherry-picked the relevant commits on this branch:
https://github.com/blino/WebKitForWayland/tree/no-opengl

Some are pretty generic, and I'll push them upstream.
Some are still work in progress or ugly hacks, and @zdobersek will probably not like them ;-)
Especially the DrawingAreaImpl stuff.

You have to build with `-DENABLE_OPENGL=OFF`.

Rendering currently works fine internally, I have checked doing cairo snapshots.
But there is no output to the display device.

I plan to use cairo-directfb for this.

**Comment by lazarkovacevic7** [comment]
Hello @blino,
I've used your commits (modiefied slightly) on top of stable branch and made WPE build without opengl. Although I had to turn off some additional flags (ENABLE_OPENGL, ENABLE_ACCELERATED_2D_CANVAS, USE_TEXTURE_MAPPER, USE_TEXTURE_MAPPER_GL, ENABLE_WEBGL, ENABLE_WEBGL2). My goal is to make WPE use software rendering only. So, currently I'm looking how to get something on screen with cairo+directfb.

Have you made any progress lately here?
